From 71d56c87a2fdbcc600e39867d462ea40ca4260ad Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Tue, 9 Apr 2024 07:33:11 +0000
Subject: [PATCH 05/10] bnxt_en: Drop redundant
 pci_enable_pcie_error_reporting()

This commit accommodates commit 7ec4b34be423 ("PCI/AER: Unexport
pci_enable_pcie_error_reporting()") merged in the v6.6 development cycle,
by applying changes similar to the ones found in commit 5f29b73d4eba
("bnxt: Drop redundant pci_enable_pcie_error_reporting()
") to resolve the bnxt_en driver compilation failures encountered with
the v6.6 kernel, which are listed below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=5f29b73d4eba

Resolved compilation errors:

bnxt_en-1.10.2-227.0.130.0/bnxt.c:16658:2: error: implicit declaration\
 of function pci_enable_pcie_error_reporting \
 [-Werror=implicit-function-declaration]
16658 |  pci_enable_pcie_error_reporting(pdev);
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bnxt_en-1.10.2-227.0.130.0/bnxt.c: In function bnxt_remove_one:
bnxt_en-1.10.2-227.0.130.0/bnxt.c:17632:2: error: implicit declaration\
 of function pci_disable_pcie_error_reporting \
 [-Werror=implicit-function-declaration]
17632 |  pci_disable_pcie_error_reporting(pdev);

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 bnxt_en-1.10.2-227.0.130.0/bnxt.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/bnxt_en-1.10.2-227.0.130.0/bnxt.c b/bnxt_en-1.10.2-227.0.130.0/bnxt.c
index b763655..1ccdb79 100644
--- a/bnxt_en-1.10.2-227.0.130.0/bnxt.c
+++ b/bnxt_en-1.10.2-227.0.130.0/bnxt.c
@@ -16651,8 +16651,6 @@ static int bnxt_init_board(struct pci_dev *pdev, struct net_device *dev)
 		goto init_err_release;
 	}
 
-	pci_enable_pcie_error_reporting(pdev);
-
 	INIT_WORK(&bp->sp_task, bnxt_sp_task);
 	INIT_DELAYED_WORK(&bp->fw_reset_task, bnxt_fw_reset_task);
 
@@ -17625,7 +17623,6 @@ static void bnxt_remove_one(struct pci_dev *pdev)
 		devlink_port_type_clear(&bp->dl_port);
 #endif
 	bnxt_ptp_clear(bp);
-	pci_disable_pcie_error_reporting(pdev);
 	unregister_netdev(dev);
 	bnxt_dbr_exit(bp);
 	clear_bit(BNXT_STATE_IN_FW_RESET, &bp->state);
-- 
2.43.0

